cmd/trust/keyset.go

```diff
@@ -253,3 +302,146 @@ func doListKeysets(ctx *cli.Context) error {
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		return err
+	}
+	numEntries := len(entries)
+	for idx, entry := range entries {
+		entryPath := filepath.Join(path, entry.Name())
+		padding := ""
+		if level == 0 {
+			padding = firstSym
+			if idx == (numEntries - 1) { // last one
+				padding = lastSym
+			}
+		} else {
+			// level 1 or more
+			sym := columnSym
+			if index == (total - 1) {
+				sym = " "
+			}
+			padding = fmt.Sprintf("%s%s%s", sym, strings.Repeat(" ", (level*4)-1), firstSym)
+			if idx == (numEntries-1) || entry.IsDir() { // will recurse
+				padding = fmt.Sprintf("%s%s%s", sym, strings.Repeat(" ", (level*4)-1), lastSym)
+			}
+		}
+		if entryPath != prevpath {
+			entryText := entry.Name()
+			if entry.IsDir() {
+				entryText = color.BlueString("%s", entryText)
+			}
+			fmt.Printf("%s %s\n", padding, entryText)
+		}
+		if entry.IsDir() {
+			newLevel := level + 1
+			return printDir(entryPath, path, newLevel, index, total)
+		}
+	}
+	return nil
+}
+
+func doShowKeyset(ctx *cli.Context) error {
+	if len(ctx.Args()) == 0 {
+		return fmt.Errorf("Please specify keyset name. Select from 'trust keyset list'")
+	}
+
+	keysetName := ctx.Args()[0]
+	if keysetName == "" {
+		return fmt.Errorf("Please specify keyset name. Select from 'trust keyset list'")
+	}
+
+	moskeysetPath, err := getMosKeyPath()
+	if err != nil {
+		return err
+	}
+
+	keysetPath := filepath.Join(moskeysetPath, keysetName)
+	if !PathExists(keysetPath) {
+		return fmt.Errorf("Unknown keyset '%s', cannot find keyset at path: %q", keysetName, keysetPath)
+	}
+
+	// no keyset key name specified only, print path if --path, otherwise list all key dir names
+	if len(ctx.Args()) < 2 {
+		fmt.Printf("%s\n", color.BlueString(keysetPath))
+		for idx, keyDir := range KeysetKeyDirs {
+			keyPath := filepath.Join(keysetPath, keyDir)
+			if ctx.Bool("path") {
+				fmt.Printf("%s\n", color.BlueString(keyPath))
+				continue
+			}
+			padding := firstSym
+			if idx == (len(KeysetKeyDirs) - 1) {
+				padding = lastSym
+			}
+			fmt.Printf("%s %s\n", padding, color.BlueString(keyDir))
+			if err := printDir(keyPath, "", 1, idx, len(KeysetKeyDirs)); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	keyName := ctx.Args()[1]
+	if keyName == "" {
+		return fmt.Errorf("Please specify keyset key name, must be one of: %s", strings.Join(KeysetKeyDirs, ", "))
+	}
+
+	if !isValidKeyDir(keyName) {
+		return fmt.Errorf("Invalid keyset key name '%s':, must be one of: %s", strings.Join(KeysetKeyDirs, ", "))
+	}
+
+	keyPath := filepath.Join(keysetPath, keyName)
+	if !PathExists(keyPath) {
+		return fmt.Errorf("Keyset %s key %q does not exist at %q", keysetName, keyName, keyPath)
+	}
+
+	if len(ctx.Args()) > 2 {
+		item := ctx.Args()[2]
+		fullPath := filepath.Join(keyPath, item)
+		if !PathExists(fullPath) {
+			return fmt.Errorf("Failed reading keyset %s key %s item %s at %q: %w", keysetName, keyName, item, fullPath, err)
+		}
+
+		contents, err := ioutil.ReadFile(fullPath)
```

Comment:
Note that ioutil.ReadFile() is deprecated in favor of os.ReadFIle().  By swapping that, you can avoid the new import above.

Reply:
OK, will fix